weave/trace_server/refs_internal.py

```diff
@@ -29,23 +31,8 @@ class InvalidInternalRef(ValueError):
     pass
 
 
-def quote_select(s: str, quote_chars: tuple[str, ...] = ("/", ":", "%")) -> str:
-    """We don't need to quote every single character, rather
-    we just need to quote the characters that are used as
-    delimiters in the URI. Right now, we only use "/", ":", and "%". Moreover,
-    the only user-controlled fields are the object name and the extra fields.
-
-    All other fields are generated by the system and are safe for parsing.
-    """
-    # We have to be careful here since quoting creates new "%" characters.
-    # We don't want to double-quote characters.
-    if "%" in quote_chars and quote_chars[-1] != "%":
-        raise ValueError("Quoting '%' must be the last character in the list.")
-
-    for c in quote_chars:
-        s = s.replace(c, urllib.parse.quote(c))
-
-    return s
+def extra_value_quoter(s: str) -> str:
```

Comment:
<img width="819" alt="Screenshot 2024-08-27 at 3 37 44 PM" src="https://github.com/user-attachments/assets/b02889a4-509e-4c15-a68a-7aa84c630bd9">
consider including a note on which chars this does not quote